src/commands/package/version/create/list.ts

```diff
@@ -80,9 +112,53 @@ export class PackageVersionCreateListCommand extends SfCommand<CreateListCommand
           header: messages.getMessage('createdBy'),
         },
       };
+      if (flags.verbose) {
+        try {
+          await this.addVerboseData(results, columnData);
+        } catch (err) {
+          const errMsg = typeof err === 'string' ? err : err instanceof Error ? err.message : 'unknown error';
+          this.warn(`error when retrieving verbose data (package name and version) due to: ${errMsg}`);
+        }
+      }
       this.table(results, columnData, { 'no-truncate': true });
     }
 
     return results;
   }
+
+  // Queries Package2Version for the name and version number of the packages and adds that data
+  // to the results and table output.
+  private async addVerboseData(results: CreateListCommandResult, columnData: ColumnData): Promise<void> {
+    // Query for the version name and number data
+    const queryFields = ['Id', 'Name', 'MajorVersion', 'MinorVersion', 'PatchVersion', 'BuildNumber'];
+    const whereclause = "WHERE Id IN ('%IDS%')";
+    const whereClauseItems = results.map((pvcrr) => pvcrr.Package2VersionId).filter(Boolean);
+    type PackageVersionData = Pick<
+      PackagingSObjects.Package2Version,
+      'Id' | 'Name' | 'MajorVersion' | 'MinorVersion' | 'PatchVersion' | 'BuildNumber'
+    >;
+    type VersionDataMap = {
+      [id: string]: { name: string; version: string };
+    };
+    const versionData = await PackageVersion.queryPackage2Version<PackageVersionData>(
+      this.connection,
+      queryFields,
+      whereclause,
+      whereClauseItems
+    );
+    const vDataMap: VersionDataMap = {};
+    versionData.map((vData) => {
+      const version = getPackageVersionNumber(vData as PackagingSObjects.Package2Version, true);
+      vDataMap[vData.Id] = { name: vData.Name, version };
+    });
```

Comment:
this *should* let you remove the internal type definition for VersionDataMap since the object type becomes inferable when you avoid mutating the empty object.
 
```suggestion
    const vDataMap = Object.fromEntries(
      versionData.map((vData) => {
        const version = getPackageVersionNumber(vData as PackagingSObjects.Package2Version, true);
        return [vData.Id, { name: vData.Name, version }]
      });
    )
```